### PR TITLE
Adding support for debug logs in executed functions.

### LINKED ISSUE #735: [BUG] Logging level debug does not work in local or portal logs
###### Please provide the following:

In any python function, log level debug does not display in both local and portal logs.

###### Provide any related information

* Links to source
* Bindings used

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
Changing the default log level to `Debug`. The [configuration of the log levels](https://docs.microsoft.com/en-us/azure/azure-functions/functions-host-json#logging) is set at the host.json.

FYI @stefanushinardi @anthonychu 

Fixes #735

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure_functions_worker/dispatcher.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ async def dispatch_forever(self):
             # established, should use it for system and user logs
             logging_handler = AsyncLoggingHandler()
             root_logger = logging.getLogger()
-            root_logger.setLevel(logging.INFO)
+            root_logger.setLevel(logging.DEBUG)
```
Comment (vrdmr) [inline azure_functions_worker/dispatcher.py:125]: the core change - INFO to DEBUG.

File: azure_functions_worker/main.py [hunk 2]
```diff
@@ -5,21 +5,28 @@
...
+                        help='id for the worker')
+    parser.add_argument('--requestId', dest='request_id',
+                        help='log destination: stdout, stderr, '
+                             'syslog, or a file path')
     parser.add_argument('--log-level', type=str, default='INFO',
-                        choices=['TRACE', 'INFO', 'WARNING', 'ERROR'],)
+                        choices=['TRACE', 'INFO', 'WARNING', 'ERROR'],
+                        help="log level: 'TRACE', 'INFO', 'WARNING', "
+                             "or 'ERROR'")
     parser.add_argument('--log-to', type=str, default=None,
```
Comment (vrdmr) [inline azure_functions_worker/main.py:30]: Just improving the help documents.

**Comment by Hazhzeng** [comment]
I remember @ankitkumarr and I once dealt with a CRI where a customer's application performance got impacted due to too many logs from azure-storage-blob.

Since we're now enabling debug in Python in the root logger, I expect to see a bit more logger.debug from azure-storage-blob. (e.g. https://github.com/Azure/azure-storage-python/blob/4e03ff8c9d6c76a38666a8ceb8937593e93fb7eb/azure-storage-common/azure/storage/common/_auth.py#L93.)

I think it would be great if you can create some test cases with a host.json that controls the log level, and see if the host actually filters our worker logs.
https://docs.microsoft.com/en-us/azure/azure-functions/functions-host-json#sample-hostjson-file
```json
    "logging": {
        "logLevel": {
          "Function.MyFunction": "Information",
          "default": "None"
        }
```

**Comment by vrdmr** [comment]
>I think it would be great if you can create some test cases with a host.json that controls the log level, and see if the host actually filters our worker logs.
https://docs.microsoft.com/en-us/azure/azure-functions/functions-host-json#sample-hostjson-file

Done - added a new set of tests with Information log level (default) set to confirm debug logs are not sent.

File: tests/unittests/test_log_filtering_functions.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    def test_debug_logging(self):
+        r = self.webhost.request('GET', 'debug_logging')
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.text, 'OK-debug')
+
+    def check_log_debug_logging(self, host_out: typing.List[str]):
+        self.assertIn('logging info', host_out)
+        self.assertIn('logging warning', host_out)
+        self.assertIn('logging error', host_out)
+        self.assertNotIn('logging debug', host_out)
```
Comment (Hazhzeng) [inline tests/unittests/test_log_filtering_functions.py:53]: Great, this explicitly test the scenario on the host logging control side. Thanks for making the change

**Review by Hazhzeng — APPROVED** [review]
:shipit:

**Pull request merged**